app/bundles/LeadBundle/EventListener/DashboardSubscriber.php

```diff
@@ -459,5 +463,70 @@ public function onWidgetDetailGenerate(WidgetDetailEvent $event)
 
             return;
         }
+
+        if ('segments.build.time' == $event->getType()) {
+            if (!$event->isCached()) {
+                $params = $event->getWidget()->getParams();
+
+                if (empty($params['limit'])) {
+                    // Count the list limit from the widget height
+                    $limit = round((($event->getWidget()->getHeight() - 80) / 35) - 1);
+                } else {
+                    $limit = $params['limit'];
+                }
+
+                $segments = $this->leadListModel->getSegmentsBuildTime($limit, $params['order'] ?? 'desc', $params['segments'] ?? [], $canViewOthers);
+                $items    = [];
+
+                // Build table rows with links
+                if ($segments) {
+                    foreach ($segments as $segment) {
+                        $listUrl    = $this->router->generate('mautic_segment_action', ['objectAction' => 'view', 'objectId' => $segment->getId()]);
+
+                        $time = $segment->getLastBuiltTime();
+                        if ($time >= 60) {
+                            $timeString = $this->translator->trans('mautic.core.date.minute', ['%count%' => floor($time / 60)]);
+                            if ($time % 60 > 0) {
+                                $timeString .= ' '.$this->translator->trans('mautic.core.date.second', ['%count%' => round($time % 60)]);
+                            }
+                        } elseif ($time >= 1) {
+                            $timeString =  $this->translator->trans('mautic.core.date.second', ['%count%' => round($time)]);
+                        } else {
+                            $timeString =  $this->translator->trans('mautic.core.date.less.than.second');
+                        }
```

Comment:
How about this instead?
```suggestion
                        $timeString = $this->dateHelper->formatRange(
                            DateInterval::createFromDateString("{$segment->getLastBuiltTime()} seconds")
                        );
                        
```
The `DateHelper` would have to be added as a dependency to this subscriber.

Reply:
Looks much better now, but I had to group the minutes anyway because `DateInterval` doesn't convert them automatically.